### PR TITLE
Fix the focus effect of categories

### PR DESCRIPTION
**Pull request by rezkiy37**
### Details

The PR is fixing a regression from https://github.com/Expensify/App/pull/26501. There was an inconsistency between selected item and focused one. The PR syncs the focus effect with a selected item or set the focus effect to the first item.

### Fixed Issues

$ https://github.com/Expensify/App/issues/27354
PROPOSAL: N/A

#### 1. The focus effect works for the long category list
1. Open a workspace chat where there are categories more than 7.
2. Click on "**+**" _(**Actions**)_ to the left of **Composer**.
3. Click on "**Request money**".
4. Enter any amount.
5. Click on "**Next**".
6. Click on "**Show more**".
7. Click on "**Category**".
8. Verify that the first category is highlighted.
9. Press on your keyboard up/down arrows.
10. Verify that a next highlighted category is correct.
11. Click/Press "Enter" to select a category.
12. Click on "**Category**".
13. Try to use search to have a selected category within a list.
14. Verify that only the selected category is highlighted.
15. Press on your keyboard up/down arrows.
16. Verify that a next highlighted category is correct.
17. Try to use search to not have a selected category within a list.
18. Verify that the first category is highlighted.
19. Press on your keyboard up/down arrows.
20. Verify that a next highlighted category is correct.

#### 2. The focus effect works for the short category list
1. Open a workspace chat where there are categories less than 8.
2. Click on "**+**" _(**Actions**)_ to the left of **Composer**.
3. Click on "**Request money**".
4. Enter any amount.
5. Click on "**Next**".
6. Click on "**Show more**".
7. Click on "**Category**".
8. Verify that the first category is highlighted.
9. Press on your keyboard up/down arrows.
10. Verify that a next highlighted category is correct.
11. Click/Press "Enter" to select a category.
12. Click on "**Category**".
13. Verify that only the selected category is highlighted.
14. Press on your keyboard up/down arrows.
15. Verify that a next highlighted category is correct.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as "**Tests**".
### QA Steps

Same as "**Tests**".
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/57314631/8c53c053-8282-4af7-a83b-2d2dee19428b

https://github.com/Expensify/App/assets/57314631/13f68b49-4ea2-43e1-aeda-0113a6685b7d

https://github.com/Expensify/App/assets/57314631/9026c2bc-4e12-4d94-9ddc-5891e2d02977

https://github.com/Expensify/App/assets/57314631/082e9553-e484-43f5-8933-e9cd4da27a4c

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/57314631/53df3e5b-b772-44b2-aaff-08a51c017ad5

https://github.com/Expensify/App/assets/57314631/c7ec1c0c-1354-4dcd-ae88-af0052ad0ad5

https://github.com/Expensify/App/assets/57314631/cd6c6349-ea77-4bf5-98ba-281fa5712a99

https://github.com/Expensify/App/assets/57314631/f4ee4987-c547-4f80-b3ad-f0e3b697195b

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/57314631/7e3d3fe4-0be8-4147-bd89-2af72f664a70

https://github.com/Expensify/App/assets/57314631/db1b94d2-8e94-4099-a010-ada60c1bea3f

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/57314631/a6dadddb-7d21-4066-ac39-ad6acedf829c

https://github.com/Expensify/App/assets/57314631/8e655b56-cc57-4d56-b0f8-d195b85c6ea8

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/57314631/1a3ee3cc-b8d6-47be-8b14-8c72b0d1281c

https://github.com/Expensify/App/assets/57314631/499c1acd-c391-4fb1-9094-0698a9cf64a6

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/57314631/1961b829-1446-4dff-bc28-57f6144221fb

https://github.com/Expensify/App/assets/57314631/21ffebdc-45c1-4fc9-b175-f2c3860fe909

</details>

### CODE REVIEW

**Review by burczu — APPROVED** [review]
Reviewed, tested and approved.

**Review by bondydaa — APPROVED** [review]

**Comment by kavimuru** [comment]
@bondydaa @burczu What is the size of the screen to be tested for web.chrome.large and web.chrome.small? Also for the native apps.

**Comment by chiragsalian** [comment]
Replied on slack [here](https://expensify.slack.com/archives/C9YU7BX5M/p1695850540788219)

File: src/components/CategoryPicker/index.js [hunk 1]
```diff
@@ -34,21 +34,36 @@ function CategoryPicker({policyCategories, reportID, iouType, iou, policyRecentl
         ];
     }, [iou.category]);
 
+    const sections = useMemo(() => {
```
Comment (alitoshmatov) [inline src/components/CategoryPicker/index.js:37]: Disabled category is not indicated - https://github.com/Expensify/App/issues/36485#issuecomment-1943884263

**Pull request merged**